lms/djangoapps/courseware/views/index.py

```diff
@@ -152,34 +148,25 @@ def get(self, request, course_id, chapter=None, section=None, position=None):
 
                 self.view = STUDENT_VIEW
 
-                # Do the enrollment check if enable_unenrolled_access is not enabled.
                 self.course = get_course_with_access(
                     request.user, 'load', self.course_key,
                     depth=CONTENT_DEPTH,
-                    check_if_enrolled=not self.enable_unenrolled_access,
```

Comment:
To be pedantic here, this used to say "check if the user is enrolled only if unenrolled access isn't enabled"

Our logic in `check_course_access` now takes the unenrolled access flag into account, so we don't need to worry about it here.  If it's turned on, then the `check_enrollment` function will ease permissions accordingly, if the user isn't allowed in because they're unenrolled, we'll generate a redirect correctly.